source/common/http/codec_client.cc

```diff
@@ -138,6 +138,10 @@ void CodecClient::onData(Buffer::Instance& data) {
       host_->cluster().stats().upstream_cx_protocol_error_.inc();
     }
   }
+
+  // All data should be consumed at this point.
+  ASSERT(data.length() == 0 || connection_->state() != Network::Connection::State::Open,
+         absl::StrCat("number of bytes ", data.toString()));
```

Comment:
Not sure if you meant:
 absl::StrCat("stray data ", data.toString()));
or
 absl::StrCat("number of bytes ", data.length()));